File: Backend/EdushareBackend/Entities/Dtos/Material/MaterialViewDto.cs
using Entities.Dtos.Content;
using Entities.Dtos.User;
using Entities.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Dtos.Material
{
    public class MaterialViewDto
    {
        public string Id { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public Entities.Models.Subject Subject { get; set; } = null!;
        public string Description { get; set; } = string.Empty;
        public bool IsRecommended { get; set; } = false;
        public bool IsExam { get; set; } = false;
        public DateTime UploadDate { get; set; } = DateTime.Now;
        public AppUserMaterialShortViewDto Uploader { get; set; } = null!;
        public ContentViewDto Content { get; set; } = null!;
        public List<MaterialShortViewDto> RecommendedMaterials { get; set; } = null!;
    }
}

File: Frontend/eduShare/src/app/dtos/material-view-dto.ts
import { FileContent } from "../models/file-content"
import { Subject } from "../models/subject"
import { UploaderDto } from "./uploader-dto"
import { MaterialShortViewDto } from "./material-short-view-dto"

export interface MaterialViewDto {
    id: string
    title: string
    subject: Subject
    description: string
    isRecommended:boolean
    uploadDate: string
    uploader: UploaderDto
    content: FileContent
    isExam: boolean
    averageRating: number
    ratingCount: number
    downloadCount: number
    recommendedMaterials?: MaterialShortViewDto[]
}
